src/sai/mod.rs

```diff
@@ -122,34 +144,101 @@ macro_rules! sai_hal {
                     }
                 }
 
+                /// Start listening for `event` on a given `channel`
+                pub fn listen(&mut self, channel: SaiChannel, event: Event) {
+                    let ch = match channel {
+                        SaiChannel::ChannelA => &self.rb.cha,
+                        SaiChannel::ChannelB => &self.rb.chb,
+                    };
+                    match event {
+                        Event::Overdue              => ch.im.write(|w| w.ovrudrie().set_bit()),
+                        Event::Muted                => ch.im.write(|w| w.mutedetie().set_bit()),
+                        Event::WrongClock           => ch.im.write(|w| w.wckcfgie().set_bit()),
+                        Event::Data                 => ch.im.write(|w| w.freqie().set_bit()),
+                        Event::AnticipatedFrameSync => ch.im.write(|w| w.afsdetie().set_bit()),
+                        Event::LateFrameSync        => ch.im.write(|w| w.lfsdetie().set_bit()),
```

Comment:
I happened to be looking at this again, should these not be `modify` rather than `write`? I would expect multiple calls to `listen()` to set all the events, not just the last one.

Reply:
What's the difference? Sounds like modify should be for changed vs ones that are just written like flags?

Reply:
Yes, each flag set requires a call to listen, what do you mean? The example only registers the data flag currently.

Reply:
Ah, modify specifically refers to "Read-Modify-Write" - [svd2rust explains it well](https://en.wikipedia.org/wiki/Read%E2%80%93modify%E2%80%93write). The outcome for `write` is that all the bits not in the closure are cleared.

Reply:
To clarify with an example, I would expect

```rust
let mut sai = ...;

sai.listen(Event::Data);
sai.listen(Event::WrongClock);
```

to set both the Data and WrongClock flags.